osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -116,25 +120,73 @@ void handleHitCircle(DrawableHitCircle circle)
             }
         }
     }
+
+    public class OsuModRelax : ModRelax, IOsuModRelaxExtensions
+    {
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Concat(new[] { typeof(OsuModAutopilot), typeof(OsuModMagnetised), typeof(OsuModAlternate), typeof(OsuModSingleTap) }).ToArray();
+        public override LocalisableString Description => @"You don't need to click. Give your clicking/tapping fingers a break from the heat of things.";
+
+        public bool IsDownState { get; set; }
+        public bool WasLeft { get; set; }
+        public OsuInputManager InputManager { get; set; } = null!;
+        public ReplayState<OsuAction> State { get; set; } = null!;
+        public double LastStateChangeTime { get; set; }
+        public bool HasReplay { get; set; }
+
+        // These are required because these methods fulfill other interfaces.
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset) => OsuModRelaxExtensions.ApplyToDrawableRuleset(this, drawableRuleset);
+        public void ApplyToPlayer(Player player) => OsuModRelaxExtensions.ApplyToPlayer(this, player);
+        public void Update(Playfield playfield) => OsuModRelaxExtensions.Update(this, playfield);
+
+        public OsuAction LeftAction => OsuAction.LeftButton;
+        public OsuAction RightAction => OsuAction.RightButton;
+    }
+
+    public class OsuModRelaxDisplay : ModRelaxDisplay, IOsuModRelaxExtensions, IApplicableToDrawableHitObject
+    {
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Concat(new[] { typeof(OsuModAutopilot), typeof(OsuModMagnetised), typeof(OsuModAlternate), typeof(OsuModSingleTap) }).ToArray();
+        public override LocalisableString Description => @"It looks like you have relax on, but your clicks are still scored. Just focus on your aim - and make sure to tap perfectly.";
+
+        public bool IsDownState { get; set; }
+        public bool WasLeft { get; set; }
+        public OsuInputManager InputManager { get; set; } = null!;
+        public ReplayState<OsuAction> State { get; set; } = null!;
+        public double LastStateChangeTime { get; set; }
+        public bool HasReplay { get; set; }
+
+        // These are required because these methods fulfill other interfaces.
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset) => OsuModRelaxExtensions.ApplyToDrawableRuleset(this, drawableRuleset);
+        public void ApplyToPlayer(Player player) => OsuModRelaxExtensions.ApplyToPlayer(this, player);
+        public void Update(Playfield playfield) => OsuModRelaxExtensions.Update(this, playfield);
+
+        public OsuAction LeftAction => OsuAction.UnscoredLeftButton;
+        public OsuAction RightAction => OsuAction.UnscoredRightButton;
+
+        public void ApplyToDrawableHitObject(DrawableHitObject drawable)
+        {
+            if (!HasReplay)
+                drawable.DisplayScoredResult = false;
+        }
```

Comment:
I couldn't figure out if there was a better way to do this similar to what `OsuModStrictTracking` does. This might be preferable anyway by not overwriting the whole object, making it compatible with more (future) mods.